### DISCUSSION TITLE
Docker socket is not available.

### DISCUSSION THREAD

**Discussion by vibraphone** [nextcloud/all-in-one#485] [Questions] [answered]
Hi all, although #10 is closed, I appear to be having the same problem when running the docker container for the first time on Fedora 35:

```text
# docker run -it \   
--name nextcloud-aio-mastercontainer \
--restart always \
-p 80:80 \
-p 8080:8080 \
-p 8443:8443 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest 
Docker socket is not available. Cannot continue.
```

**Comment by szaimen** [comment]
Hi, did you install docker using the recommended command?
`curl -fsSL get.docker.com | sudo sh`

**Reply by vibraphone** [reply]
No, I would prefer to use the system packages that get regular security updates. Also, I already have the system packages installed and am running other images.

**Reply by szaimen** [reply]
then you need to figure out where the docker socket is located on your system and use that for the command but I will not be able to help you on this.

**Reply by vibraphone** [reply]
Thanks.

For others who may find this: I suspect it is selinux preventing things from working; the docker socket is in `/var/run/docker.sock` just as the instructions assume. Indeed, running like so (note the `:z` options)  gets a little further:

```text
docker run -it \                                   
--name nextcloud-aio-mastercontainer \
--restart always \
-p 80:80 \
-p 8080:8080 \
-p 8443:8443 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config:z \
--volume /var/run/docker.sock:/var/run/docker.sock:z,ro \
nextcloud/all-in-one:latest
Trying to fix docker.sock permissions internally...
Creating docker group internally with id 976
Cannot connect to the docker socket. Cannot proceed.
```

**Reply by zegang** [reply]
Any way found to resolve this ?

**Reply by Zoey2936** [reply]
do you use SE linux? if yes can you try to add `--privileged` to the start command, please?

**Reply by zegang** [reply]
> do you use SE linux? if yes can you try to add `--privileged` to the start command, please?

It works. Thank you @Zoey2936

**Reply by Zoey2936** [reply]
@szaimen I've got this from the portainer docs, SE-Linux Systems require this option because access to the docker socket is a privileged access on SE-Linux

**Reply by szaimen** [reply]
So it cannot be even made work with `/var/run/docker.sock:/var/run/docker.sock:z,ro` if SELinux is enabled?

**Reply by Zoey2936** [reply]
no

**Reply by Zoey2936** [reply]
@zegang can you please also try this: https://github.com/dpw/selinux-dockersock/issues/8#issuecomment-928153181 so adding `--security-opt label=disabled` instead of privilleged?

**Reply by zegang** [reply]
> @zegang can you please also try this: [dpw/selinux-dockersock#8 (comment)](https://github.com/dpw/selinux-dockersock/issues/8#issuecomment-928153181) so adding `--security-opt label=disabled` instead of privilleged?

It works with '--security-opt label=disable'.
```
$ sudo docker run -it --name nextcloud-aio-mastercontainer --restart always -p 80:80 -p 8080:8080 -p 8443:8443 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config:z --volume /var/run/docker.sock:/var/run/docker.sock:ro --security-opt label=disable nextcloud/all-in-one:latest
Trying to fix docker.sock permissions internally...
Creating docker group internally with id 974
Initial startup of Nextcloud All-in-One complete!
You should be able to open the Nextcloud AIO Interface now on port 8080 of this server!
E.g. https://internal.ip.of.this.server:8080

If your server has port 80 and 8443 open and you point a domain to your server, you can get a valid certificate automatically by opening the Nextcloud AIO Interface via:
https://your-domain-that-points-to-this-server.tld:8443
{"level":"info","ts":1685955508.1206856,"msg":"using provided configuration","config_file":"/Caddyfile","config_adapter":""}
{"level":"info","ts":1685955508.1247776,"msg":"failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Receive-Buffer-Size for details."}
[05-Jun-2023 08:58:29] NOTICE: fpm is running, pid 116
[05-Jun-2023 08:58:29] NOTICE: ready to handle connections
```

**Reply by Zoey2936** [reply]
@szaimen I think this needs to be added to AIO in some way

**Reply by szaimen** [reply, marked as answer]
Yes, I am adjusting things:
https://github.com/nextcloud/all-in-one/pull/2670

**Reply by szaimen** [reply]
@zegang is `--volume /var/run/docker.sock:/var/run/docker.sock:ro` without the z working fine if one uses `--security-opt label=disabled` ?

**Reply by zegang** [reply]
> @zegang is `--volume /var/run/docker.sock:/var/run/docker.sock:ro` without the z working fine if one uses `--security-opt label=disabled` ?

Yes. How to add 'security_opt' in docker-compose yaml file ?

**Reply by szaimen** [reply]
Probably something like
```
    security_opt:
      - label=disabled
```

**Comment by agent-random** [comment]
Nextcloud All-in-One: A Security Compromise in Production
Nextcloud AIO disables SELinux protection via security_opt: label=disabled, creating significant security weaknesses:
1. Violates Defense in Depth
 Containers are designed to provide isolation. Disabling SELinux defeats this purpose, leaving the host exposed if a container is compromised.
2. False Sense of Security
 Documentation suggests SELinux configuration, yet AIO overrides it silently. Users following best practices get no real protection.
3. No Proper Docker-in-Docker SELinux Policy
 AIO requires high privileges to spawn child containers but provides no SELinux policy to allow this securely. The project acknowledges the limitation but offers no fix.
4. Production Risk
 Deploying AIO in production with label=disabled means:
- Container can read/write any host file
- No SELinux enforcement between containers
- Exposed attack surface if container is breached
5. Misleading Documentation
 The SELinux configuration guide applies to traditional installations only. AIO users are left without guidance.
Conclusion
 Nextcloud AIO is unsuitable for production environments requiring SELinux enforcement. Either implement proper Docker-in-Docker SELinux policies or use traditional deployment.
